modules/next/src/Plugin/Next/Revalidator/CacheTag.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\next\Plugin\Next\Revalidator;
+
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Utility\Error;
+use Drupal\next\Event\EntityActionEvent;
+use Drupal\next\Plugin\ConfigurableRevalidatorBase;
+use Drupal\next\Plugin\RevalidatorInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a revalidator for revalidating Next Site references by cache tags.
+ *
+ * @Revalidator(
+ *  id = "cache_tag",
+ *  label = "Cache Tag",
+ *  description = "Cache tag based on-demand revalidation."
+ * )
+ */
+class CacheTag extends ConfigurableRevalidatorBase implements RevalidatorInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    // No configuration form.
+  }
+
+  /**
+   * {@inheritdoc}
+   * @throws \Drupal\Core\Entity\EntityMalformedException
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function revalidate(EntityActionEvent $event): bool {
+    $revalidated = FALSE;
+    $entity = $event->getEntity();
+    $sites = $event->getSites();
+
+    if (!($entity instanceof FieldableEntityInterface)) {
+      return FALSE;
+    }
+
+    $tags = [];
+    $tags[] = $entity->getEntityTypeId() . ':' . $entity->id();
```

Comment:
```suggestion
$tags = $entity->getCacheTags();
```

Hi! how would you feel about using the getCacheTags() method here instead? Seems like it would have the same outcome?

![image](https://github.com/user-attachments/assets/920abc33-4add-4f6d-b380-27b0ab223ad8)

Reply:
Good suggestion. I just updated it in 59281a977563c37e5fbebbe350940f0d4ce68bf0 I am not sure if getCacheTags is available in all entities (including custom entities). What do you think ? @robdecker @backlineint

Reply:
Good question! I had a look and I think it should be there on all entities because: 

1. the method appears on the `CacheableDependencyInterface` https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Cache/CacheableDependencyInterface.php#L42
1. the `EntityInterface` which should be used for all entities extends the `CacheableDependencyInterface` https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Core/Entity/EntityInterface.php#L14

Reply:
Fixed in a38006f3722292d34ba35969c4f46cb4a5677f4f